### ISSUE TITLE
Custom Losses Documentation is Incomplete

### ISSUE THREAD

**Issue by lonerzzz** [tensorflow/tensorflow#49236]
## URL(s) with the issue:

https://www.tensorflow.org/guide/keras/train_and_evaluate#custom_losses

### Clear description

The current documentation for writing custom loss functions is significantly incomplete. It does not indicate:

1) how to test compilation of the custom loss function
2) how to debug the custom loss function
3) what data types are expected for the function
4) what data type is expected to be returned from the function
5) what operations can be performed within the function.
For example. I found when run in the graph, functions like .numpy(), make_ndarray() and to_list() do not work on the tensors when in graph mode. However, these issues seem only to have arisen through raised issues rather than being proactively documented. Likewise, the documentation on how to to reenable these functions is not clearly documented.

**Comment by a-ma-n** [comment]
Hello there! is this issue open? can I contribute?

**Comment by lonerzzz** [comment]
@a-ma-n if you have any commentary, I personally would appreciate it and any knowledge can help improve the guide going forward so other can benefit

**Comment by a-ma-n** [comment]
Great I'll look into it

**Comment by Bhavay192** [comment]
Hey @a-ma-n, Are u still looking into the issue?? If not, then can I work on the same??

**Comment by MarkDaoust** [comment]
These guides come from the keras-io repo:

https://github.com/keras-team/keras-io

Be sure to talk with them before undertaking any significant work, the owners are particular about what goes there.

Also note that many of the specific questions in this bug can be answered with "the function should be executable in a `tf.function`: it must be composed of TensorFlow operations"

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
